### PR TITLE
Dev terrado2-Cancellation Feature

### PR DESCRIPTION
**Pull request by espresso64**
Cancellation feature to be merged with master branch. 

**Notes**
-Modified customer controller/service code to retrieve previous flights + reservation number
-Created ReservationFlightInfo domain, reserves flightInfo for the flights service.
-Added findByReservationid to the ReservationRepository
-Modified previous_flights.html to display reservation number and radio buttons to select flight to cancel
-Current form does not reintroduce seats back to the pool.

### CODE REVIEW

File: src/main/java/cst438flights/controller/CustomerController.java [hunk 1]
```diff
@@ -52,8 +52,8 @@ public String customerHistory(
             @RequestParam("password") String password,
             Model model) {
         Customer customer = customerRepository.findByEmail(email);
-        System.out.println("Password stored " + customer.getPassword());
-        System.out.println("Password Entered " + password);
+        //System.out.println("Password stored " + customer.getPassword());
```
Comment (Anelon) [inline src/main/java/cst438flights/controller/CustomerController.java:55]: Commented out debugging statements to clean up some console logs

File: src/main/java/cst438flights/domain/FlightInfo.java [hunk 2]
```diff
@@ -4,11 +4,11 @@
...
-    long id;
-    String departureAirport;
-    String arrivalAirport;
-    Date departureDate;
-    String status;
+    private long id;
+    private String departureAirport;
+    private String arrivalAirport;
+    private Date departureDate;
+    private String status;
```
Comment (Anelon) [inline src/main/java/cst438flights/domain/FlightInfo.java:11]: For consistency added private keyword

File: src/main/resources/templates/index.html [hunk 3]
```diff
@@ -37,7 +37,7 @@ <h3>Select your departing city</h3> <br />
 
     <form action="#" th:action="@{/customer}" method="post">
         <h3>Look up previous flights</h3>
-        <input type="text" name="email" id="email" value="email">
+        <input type="text" name="email" id="email" placeholder="email">
```
Comment (Anelon) [inline src/main/resources/templates/index.html]: Changed from value to placeholder so that it is easier to input into field

**Review by Anelon — APPROVED** [review]
Looks good, Might be worth adding the ability to cancel more than one flight at a time (and show that you have no flights if the user doesn't have any flights to cancel)

File: src/main/resources/templates/index.html [hunk 4]
```diff
@@ -37,9 +37,9 @@ <h3>Select your departing city</h3> <br />
 
     <form action="#" th:action="@{/customer}" method="post">
         <h3>Look up previous flights</h3>
-        <input type="text" name="email" id="email" value="email">
+        <label for="email">Email </label><input type="text" name="email" id="email" placeholder="email">
```
Comment (Nhinds88) [inline src/main/resources/templates/index.html:40]: yeah I agree changing to placeholder here is best

**Review by Nhinds88 — APPROVED** [review]

**Pull request merged**